### DISCUSSION TITLE
Color Threshold or Color Scale in chart

### DISCUSSION THREAD

**Discussion by octaviz** [dbuezas/lovelace-plotly-graph-card#301] [Q&A] [answered]
There would be a way to integrate "colorscales" to work in Home Assistant, in Plotly there are ways but I do not know how to adapt it, there are others like Apexchart that has the option "color_threshold:" that allows through values to make it change color depending on the value.

In Apexchart it looks like this:
![imagen](https://github.com/dbuezas/lovelace-plotly-graph-card/assets/856579/578e5c14-a3ad-403c-b68e-28db36756186)

En la web offical de Plotly hay opciones https://plotly.com/python/colorscales/

**Comment by dbuezas** [comment, marked as answer]
Plotly's colorscales are not for line charts unfortunately. There's no way to achieve this with Plotly.
Maybe this other option serves the same purpose:
#45

**Reply by octaviz** [reply]
Thank you very much, if they were "marker" type dots it is not possible either?

**Reply by dbuezas** [reply]
With markers it's possible, I think you'll need to construct an array of colors (or at least intensities). For that you can use $ex or $fn.

Obviously, this will only colorize the markers, not the lines (if you were to use lines+markers).

Another possibility is to draw three traces with the same entity and map all values outside each range to null using filters
